airflow/jobs/scheduler_job.py

```diff
@@ -672,33 +672,24 @@ def _process_task_instances(self, dag, task_instances_list, session=None):
                     self.log.debug('Queuing task: %s', ti)
                     task_instances_list.append(ti.key)
 
-    def _process_dags(self, dagbag, dags, tis_out):
+    @provide_session
+    def _process_dags(self, dags, tis_out, session=None):
```

Comment:
While touching this line, can we add type annotations?

Reply:
I want to add types for the scheduler as a separate PR. In this PR and other performance, I try to avoid additional changes.

Reply:
Adding types for classes from the core often causes cyclical imports, so we must be careful.